plugins/file/zipfile/zip_file_reader.go

```diff
@@ -0,0 +1,55 @@
+package zipfile
+
+import (
+	"archive/zip"
+	"bytes"
+	"io"
+
+	"github.com/chrislusf/gleam/util"
+)
+
+type FileReader struct {
+	reader   *zip.ReadCloser
+	Cursor   int
+	NumFiles int
+}
+
+func New(filename string) *FileReader {
+	r, _ := zip.OpenReader(filename)
+	numFilesAndDirs := len(r.File)
+	return &FileReader{
+		reader:   r,
+		Cursor:   0,
+		NumFiles: numFilesAndDirs,
+	}
+}
+
+func (r *FileReader) ReadHeader() (fieldNames []string, err error) {
+	return nil, nil
+}
+
+// Read will iterate through the zip file, it will treat each file
+// in the zipfile as a row and return back to the caller, where the
+// key is file or directory name and the value is the bytes of the
+// file from the input zip file
+func (r *FileReader) Read() (row *util.Row, err error) {
+	if r.Cursor >= r.NumFiles {
+		return nil, io.EOF
+	}
+	f := r.reader.File[r.Cursor]
+	fp, err := f.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer fp.Close()
+	data := new(bytes.Buffer)
+	data.ReadFrom(fp)
+	row = util.NewRow(util.Now())
+	if !f.FileInfo().IsDir() {
+		row.AppendKey(f.Name).AppendValue(data.Bytes())
```

Comment:
I'm not sure if this is what you mean by return the name as well, its ugly but it will work

Reply:
does the `f.Name` have the full path? If not, please try to use full path.

Reply:
yes, it's the full path to the file in the zip